### PR TITLE
[v3.0] [CI:DOCS] Cirrus: Fix running Validate task on branches

### PR DESCRIPTION
**Pull request by cevich**
Backport from master https://github.com/containers/podman/pull/9063

Followup to 4c60523

For whatever reason Cirrus-CI does not set `$CIRRUS_BASE_SHA` when
testing on a branch or tag.  If the branch happened to also not have
any successful previous runs, then `$EPOCH_TEST_COMMIT` was being set
to the error-flagging value `YOU_FOUND_A_BUG`.

Fix this by detecting an empty value when running on a branch or via
`hack/get_ci_vm.sh` (`$CIRRUS_TAG` will be empty) and setting
`$CIRRUS_BASE_SHA` based on the upstream branch HEAD.  Otherwise,
instead of using `YOU_FOUND_A_BUG`, simply reference the HEAD commit.

Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@edsantiago PTAL
@containers/podman-maintainers PTAL

**Review by QiWang19 — COMMENTED** [review]
LGTM

**Review by edsantiago — CHANGES_REQUESTED** [review]
Same feedback as #9063. This must not merge as-is.

**Comment by rhatdan** [comment]
I would prefer to merge the main branch patch, once @edsantiago comments are addressed and then cherry-pick the fix to the 3.0 branch.

**Comment by cevich** [comment]
> then cherry-pick the fix to the 3.0 branch.

okay, will do.

**Pull request closed without merging**
